### DISCUSSION TITLE
Help setting up application.properties file

### DISCUSSION THREAD

**Discussion by cdaug16** [abhi9720/BankingPortal-API#36] [Q&A] [answered]
I found the example application.properties.sample file provided with the repository. I included the default below with the updates I made to it in angle brackets <>. The "Installation and Setup" section of the readme just says "3. Configure MySQL: Set up a MySQL database, create a copy of application.properties.sample, rename it application.properties, and update the properties as needed." I was talking to @AmrElsayyad about updates to the Wiki for some changes to the API, wanted to see if there was any feedback for making this step clearer.

```
# sample application.properties file

server.port=8180
spring.datasource.url=jdbc:mysql://localhost:3306/bankingapp
spring.datasource.username=root
spring.datasource.password=<MySQL Password>
spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
spring.jpa.database-platform=org.hibernate.dialect.MySQLDialect
spring.jpa.generate-ddl=true
spring.jpa.show-sql=false
spring.jpa.hibernate.ddl-auto=update
spring.main.allow-circular-references=true
server.error.include-message=always

# JWT
jwt.secret=<Client Secret Key>
# Token expiration time in milliseconds (e.g., 24 hours)
jwt.expiration=86400000
jwt.header=Authorization
jwt.prefix=Bearer

# Spring Mail
spring.mail.host=smtp.gmail.com
spring.mail.port=587
spring.mail.username=<Personal Email>
spring.mail.password=<Personal Email Password>
spring.mail.properties.mail.smtp.auth=true
spring.mail.properties.mail.smtp.starttls.enable=true

# Geolocation API
geo.api.url=https://api.findip.net/
geo.api.key=<API Key>
```

1. Do I need to make an account with JWT.io? The application seemed to work with the default value for jwt.secret.
2. Do I need to do anything special to setup SMTP with an existing Gmail account? I found this article from an error with the email authentication from the console, so I'm going to look at this more: https://support.google.com/accounts/answer/185833?visit_id=638603178520098130-4261990959&p=InvalidSecondFactor&rd=1#zippy=%2Cwhy-you-may-need-an-app-password
3. Do I need to make an account with FindIP.net? Even after I updated the geo.api.key to my key, I still get an invalid IP when I login to a user account.

These are the errors I get from the console on user login:
```
2024-08-26 20:13:19.951  INFO [task-1] : Getting geolocation for IP: 0:0:0:0:0:0:0:1
2024-08-26 20:13:19.959  INFO [http-nio-8180-exec-4] : Generating token for user: <Account Number>
2024-08-26 20:13:20.012  INFO [http-nio-8180-exec-4] : Saving token for account: <Account Number>
2024-08-26 20:13:20.293 ERROR [task-1] : Failed to get geolocation for IP: 0:0:0:0:0:0:0:1
2024-08-26 20:13:20.673 ERROR [task-2] : Failed to send email to <User Email>
```

**Comment by cdaug16** [comment]
Question 2 Resolved: I followed the steps in the link to create an "app password" for my admin email and put in my newly generated app password for spring.mail.password value.

New behavior:

- On user login, an email is sent from Auth0 to verify the login to the user account email. I click verification link.
- Need to try registering a new user after JWT set up and observe Activity on JWT dashboard
- Email is sent on login to user account email with name and time, but location is "Unknown" maybe because of error with geolocation. Email shows from admin email.
- Checked the admin email I put in the application.properties and I see the email in the Sent folder.

```
2024-08-26 20:17:27.180  INFO [task-1] : Getting geolocation for IP: 0:0:0:0:0:0:0:1
2024-08-26 20:17:27.185  INFO [http-nio-8180-exec-1] : Generating token for user: <Account Number>
2024-08-26 20:17:27.265  INFO [http-nio-8180-exec-1] : Saving token for account: <Account Number>
2024-08-26 20:17:27.374 ERROR [task-1] : Failed to get geolocation for IP: 0:0:0:0:0:0:0:1
2024-08-26 20:17:28.479  INFO [task-2] : Sent email to <User Email>
```

Still looking for feedback on 1 & 3

**Comment by AmrElsayyad** [comment, marked as answer]
@cdaug16 

1. Just generate a random JWT secret online. It's just used to sign and verify JWTs, something like a master key for authentication.

2. You have solved this one.

3. It's normal to get Unknown location because you are deploying this locally. So when calling the geolocation API with your loopback IP, the response will be null. That will not be the case in deployment, as you will need to call the API using the public IP.
